### PR TITLE
Add ValueConvertible conformance to Swift Bool

### PR DESCRIPTION
**Pull request by rcedwards**
Addresses https://github.com/Zewo/Zewo/issues/222

### CODE REVIEW

File: Sources/SQL/Core/Value.swift [hunk 1]
```diff
@@ -207,3 +207,18 @@ extension UInt64: ValueConvertible {
         return .string(String(self))
     }
 }
+
+extension Bool: ValueConvertible {
+    public init(rawSQLData buffer: Buffer) throws {
+        let stringValue = try String(buffer: buffer)
```
Comment (Danappelxx) [inline Sources/SQL/Core/Value.swift]: Could you use `Int(buffer:)` here?

File: Sources/SQL/Core/Value.swift [hunk 2]
```diff
@@ -207,3 +207,18 @@ extension UInt64: ValueConvertible {
...
+        guard stringValue == "1" || stringValue == "0" else {
```
Comment (Danappelxx) [inline Sources/SQL/Core/Value.swift]: I would use a switch:

``` swift
switch value {
    case 0: return false
    case 1: return true
    default: throw ValueError(...)
}
```

File: Sources/SQL/Core/Value.swift [hunk 3]
```diff
@@ -207,3 +207,18 @@ extension UInt64: ValueConvertible {
...
+            throw ValueError(description: "Failed to convert data to Bool")
+        }
+
+        self = stringValue == "1" ? true : false
+    }
+
+    public var sqlValue: Value {
+        return .string(String(self ? "1" : "0"))
```
Comment (Danappelxx) [inline Sources/SQL/Core/Value.swift]: Use Int's implementation here: `(self ? 1 : 0).sqlValue`

File: Tests/SQLTests/SQLTests.swift [hunk 4]
```diff
@@ -4,6 +4,7 @@ import XCTest
 struct User {
     let username: String
     let password: String
+    let admin: Bool
```
Comment (Danappelxx) [inline Tests/SQLTests/SQLTests.swift]: `isAdmin`, maybe? Not important.

**Review by Danappelxx — CHANGES_REQUESTED** [review]
Great thanks for the pr! Just a few minor things.

**Comment by rcedwards** [comment]
Sounds good. I'll address those now.

**Review by Danappelxx — APPROVED** [review]

**Comment by Danappelxx** [comment]
Awesome! Waiting for travis to pass and I'll merge!

**Comment by rcedwards** [comment]
@Danappelxx I've discovered today that this implementation will not work with a postgres boolean field, since the buffer is returned as either "t" or "f".

Would you entertain an implementation similar to this:
```
extension Bool: ValueConvertible {
    public init(rawSQLData buffer: Buffer) throws {
        // read int, 0 = false, 1 = true
        let value = try Int(rawSQLData: Bool.sanitizedBooleanBuffer(rawBuffer: buffer))

        switch value {
        case 0: self = false
        case 1: self = true
        default: throw ValueError(description: "Failed to convert data to Bool. Excepted 0/1, got \(value).")
        }
    }

    public var sqlValue: Value {
        return (self ? 1 : 0).sqlValue
    }

    private static func sanitizedBooleanBuffer(rawBuffer: Buffer) -> Buffer {
        guard let bufferPrefix = rawBuffer.first else {
            return rawBuffer
        }
        let hex0: UInt8 = 0x30
        let hex1: UInt8 = 0x31
        let hexLowercaseT: UInt8 = 0x74
        let hexLowercaseF: UInt8 = 0x66

        switch bufferPrefix {
        case hexLowercaseT:
            return Buffer("1")
        case hexLowercaseF:
            return Buffer("0")
        case hex0, hex1:
            break
        default:
            break
        }
        return rawBuffer
    }
}
```

This works well but I could see the concern of being too database specific within the SQL framework. That being said it would be tricky to "override" this in the postgres framework since this is just an extension on bool that would collide if we tried to implement it over in https://github.com/Zewo/PostgreSQL.

Thoughts?

**Pull request merged**